File: app/src/main/java/paul/barthuel/go4lunch/ui/workmates/WorkmatesInfo.java
package paul.barthuel.go4lunch.ui.workmates;

public class WorkmatesInfo {

    private final String name;
    private final String image;
    private final String id;
    private final String placeId;
    private final String restaurantName;

    public String getRestaurantName() {
        return restaurantName;
    }

    public String getName() {
        return name;
    }

    public String getImage() { return image; }

    public String getId() { return id; }

    public String getPlaceId() { return placeId; }

    public WorkmatesInfo(String name,
                         String image,
                         String id,
                         String placeId,
                         String restaurantName) {
        this.name = name;
        this.image = image;
        this.id = id;
        this.placeId = placeId;
        this.restaurantName = restaurantName;
    }


    @SuppressWarnings("EqualsReplaceableByObjectsCall")
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        WorkmatesInfo that = (WorkmatesInfo) o;

        if (name != null ? !name.equals(that.name) : that.name != null) return false;
        if (image != null ? !image.equals(that.image) : that.image != null) return false;
        if (id != null ? !id.equals(that.id) : that.id != null) return false;
        if (placeId != null ? !placeId.equals(that.placeId) : that.placeId != null) return false;
        return restaurantName != null ? restaurantName.equals(that.restaurantName) : that.restaurantName == null;
    }

    @Override
    public int hashCode() {
        int result = name != null ? name.hashCode() : 0;
        result = 31 * result + (image != null ? image.hashCode() : 0);
        result = 31 * result + (id != null ? id.hashCode() : 0);
        result = 31 * result + (placeId != null ? placeId.hashCode() : 0);
        result = 31 * result + (restaurantName != null ? restaurantName.hashCode() : 0);
        return result;
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/workmates/WorkmatesInfo.kt
package paul.barthuel.go4lunch.ui.workmates

class WorkmatesInfo(val name: String?,
                    val image: String?,
                    val id: String?,
                    val placeId: String?,
                    val restaurantName: String?) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as WorkmatesInfo
        if (if (name != null) name != that.name else that.name != null) return false
        if (if (image != null) image != that.image else that.image != null) return false
        if (if (id != null) id != that.id else that.id != null) return false
        if (if (placeId != null) placeId != that.placeId else that.placeId != null) return false
        return if (restaurantName != null) restaurantName == that.restaurantName else that.restaurantName == null
    }

    override fun hashCode(): Int {
        var result = name?.hashCode() ?: 0
        result = 31 * result + (image?.hashCode() ?: 0)
        result = 31 * result + (id?.hashCode() ?: 0)
        result = 31 * result + (placeId?.hashCode() ?: 0)
        result = 31 * result + (restaurantName?.hashCode() ?: 0)
        return result
    }

}